src/extent.c

```diff
@@ -541,11 +541,15 @@ extents_evict(tsdn_t *tsdn, arena_t *arena, extent_hooks_t **r_extent_hooks,
 	 * the loop will iterate until the LRU extent is fully coalesced.
 	 */
 	extent_t *extent;
+	int lists = NPSIZES;
```

Comment:
Doesn't this mean we'll (effectively) never end up purging small extents so long as we can find a large one? I would think better would be to shard randomly N-ways, and cycle through the lru list we use to purge (i.e. move `lists` into the `extents_t`).

This will also reduce contention on the LRU lists if lots of threads are using a single size class often.